File: Leetcode/top_interview_questions/easy/arrays/4.contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  let temp = [];

  for (let i = 0; i < nums.length; i++) {
    if (temp.includes(nums[i])) {
      return true;
    }
    temp[i] = nums[i];
  }
  return false;
};

console.log(containsDuplicate([1, 2, 3, 4, 1, 5, 6]));

File: Leetcode/top_interview_questions/easy/arrays/4.contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
       unique = set(nums)
       
       return len(unique) != len(nums) 


sample = Solution()
print(sample.containsDuplicate([1,1,1,3,3,4,3,2,4,2]))